src/controllers/resource-store/index.tsx

```diff
@@ -49,17 +49,20 @@ export const actions: Actions = {
     getState,
     dispatch,
   }) => {
-    const { data } = getState();
-
-    const slice = getSliceForResource({ data }, { type, key });
+    const slice = getSliceForResource(getState(), { type, key });
+    const data = getNewSliceData(slice.data);
 
     dispatch(
-      setResourceState(type, key, {
-        ...slice,
-        data: getNewSliceData(slice.data),
-        expiresAt: getExpiresAt(maxAge),
-        accessedAt: getAccessedAt(),
-      })
+      data == null
```

Comment:
We are currently working on treating `null` as a valid data. Sorry for making things complicated but I am unsure that resource is removable if data is null.

Reply:
In that case it's easy to not delete the record here and just treat `null` as any other update. Either way the `promise` should resolve the `null` for consistency, which is the purpose of my PR. That said I think there needs to be some seperate semantics for a `delete()` operation.

However the bigger problem I see with `null` being valid data is that it is indistinguishable from the initial state. That is a big problem IMO for determining what state the resource is in.

Reply:
> However the bigger problem I see with null being valid data is that it is indistinguishable from the initial state. That is a big problem IMO for determining what state the resource is in.

Very true. I am examining different options. One of the solutions i think of is to set initial value to be 
```
export const BASE_DEFAULT_STATE_SLICE = {
  data: "DEFAULT_STATE_DATE",
};
```
in order to be distinguishable
What do you think?

Reply:
hmm not a fan, it needs to be a primitive type.

Really this is the whole point of `undefined` IMO.

Reply:
I think that combined with a presence of `promise` might be enough. I would need to look at the table in the `.md` file. But certainly we need to be very precise about that table going forward.

IIRC Apollo had some `linkState`(?) which makes the state explicit. I was always a fan of that since it avoids this type of problem. However I think that such a change is a bit extreme given the current API doesn't have that.